instance/models/load_balancer.py

```diff
@@ -0,0 +1,173 @@
+
+    def get_fragment_name(self):
+        """Return the base name to use for the haproxy configuration files.
+
+        This should be unique for each instance of the instance manager, so that different
+        instance managers can share a load balancing server.
+        """
+        if not self.fragment_name_postfix:
+            bits = self._meta.get_field("fragment_name_postfix").max_length * 4
+            self.fragment_name_postfix = format(random.getrandbits(bits), "x")
+            self.save()
+        return settings.LOAD_BALANCER_FRAGMENT_NAME_PREFIX + self.fragment_name_postfix
+
+    def get_config_script(self):
+        """Render the configuration script to be executed on the loader balancer."""
+        backend_map, backend_conf = self.get_configuration()
+        fragment_name = self.get_fragment_name()
+        template = loader.get_template("instance/haproxy/conf.sh")
+        return template.render(dict(
+            settings=settings,
+            conf_filename=os.path.join(settings.LOAD_BALANCER_CONF_DIR, fragment_name),
+            backend_conf=backend_conf,
+            backend_filename=os.path.join(settings.LOAD_BALANCER_BACKENDS_DIR, fragment_name),
+            backend_map=backend_map,
+        ))
+
+    def reconfigure(self):
+        """Regenerate the configuration fragments on the load-balancing server."""
+        self.logger.info("Reconfiguring load-balancing server %s", self.domain)
+        config_script = self.get_config_script()
+        try:
+            run_ssh_script(self.domain, self.ssh_username, config_script)
+        except (subprocess.CalledProcessError, subprocess.TimeoutExpired) as exc:
+            self.logger.error(
+                "Reconfiguring the load balancer failed.  Stderr of ssh process:\n%s",
+                exc.stderr,
+            )
+            raise
+
+
+def _create_default_load_balancing_server():
+    """Create the default load balancing server configured in the Django settings, if any."""
+    if settings.DEFAULT_LOAD_BALANCING_SERVER:
+        ssh_username, unused, domain = settings.DEFAULT_LOAD_BALANCING_SERVER.partition("@")
+        if not ssh_username or not domain:
+            raise ImproperlyConfigured(
+                "DEFAULT_LOAD_BALANCING_SERVER must have the form ssh_username@domain.name."
+            )
+        logger.info("Creating LoadBalancingServer %s", domain)
+        server, created = LoadBalancingServer.objects.get_or_create(
+            domain=domain,
+            defaults=dict(ssh_username=ssh_username),
+        )
+        if not created and server.ssh_username != ssh_username:
+            logger.warning(
+                "The SSH username of LoadBalancingServer %s in database does not match the "
+                "username in Django settings@ %s vs %s",
+                domain, server.ssh_username, ssh_username
+            )
+
+
+def select_load_balancing_server():
+    """Select a load-balancing server for a new instance.
```

Comment:
Hmm, the code base doesn't seem to be consistent on this, so I went with the official convention in PEP 257.  I'm happy to change it, though.